wntr/library/msx/_msxlibrary.py

```diff
@@ -0,0 +1,386 @@
+# coding: utf-8
+"""
+The wntr.msx.library module includes a library of multi-species water 
+models
+
+.. rubric:: Environment Variable
+
+.. envvar:: WNTR_LIBRARY_PATH
+
+    This environment variable, if set, will add additional folder(s) to the
+    path to search for multi-species water quality model files, 
+    (files with an ".msx", ".yaml", or ".json" file extension).
+    Multiple folders should be separated using the "``;``" character.
+    See :class:`~wntr.msx.library.ReactionLibrary` for more details.    
+"""
+
+from __future__ import annotations
+
+import json
+import logging
+import os
+from typing import Any, ItemsView, Iterator, KeysView, List, Tuple, Union, ValuesView
+
+from pkg_resources import resource_filename
+
+from wntr.msx.base import ExpressionType, ReactionType, SpeciesType
+from wntr.msx.model import MsxModel
+
+try:
+    import yaml
+
+    yaml_err = None
+except ImportError as e:
+    yaml = None
+    yaml_err = e
+
+PIPE = ReactionType.PIPE
+TANK = ReactionType.TANK
+BULK = SpeciesType.BULK
+WALL = SpeciesType.WALL
+RATE = ExpressionType.RATE
+EQUIL = ExpressionType.EQUIL
+FORMULA = ExpressionType.FORMULA
+
+logger = logging.getLogger(__name__)
+
+
+class MsxLibrary:
```

Comment:
It might be worth abstracting part of the MsxLibrary interface to allow for extensions to librarys for other purposes (eg demand pattern library). Not sure how this might look but I'd be interested in discussing it further.